[UR] Revamp README file a little (#685)
oneapi-src/unified-runtime: README.md

| Name | Description | Values | Default |
| - | - | - | - |
| UR_BUILD_TESTS | Build the tests | ON/OFF | ON |
| UR_BUILD_TOOLS | Build tools | ON/OFF | ON |
| UR_FORMAT_CPP_STYLE | Format code style | ON/OFF | OFF |
| UR_DEVELOPER_MODE | Treat warnings as errors and enables additional checks | ON/OFF | OFF |
| UR_USE_ASAN | Enable AddressSanitizer | ON/OFF | OFF |
| UR_USE_UBSAN | Enable UndefinedBehavior Sanitizer | ON/OFF | OFF |
| UR_USE_MSAN | Enable MemorySanitizer (clang only) | ON/OFF | OFF |
| UR_ENABLE_TRACING | Enable XPTI-based tracing layer | ON/OFF | OFF |

### Additional make targets

> To run automated code formatting build with option `UR_FORMAT_CPP_STYLE`

Reviewer: ```suggestion
To run automated code formatting, configure cmake with `UR_FORMAT_CPP_STYLE` option 
```

Author: Done

and then run a custom `cppformat` target:

```bash

> make cppformat

Reviewer: ```suggestion
$ make cppformat
```

```

If you've made modifications to the specification, you can also run
a custom `generate` target prior to building.
It will generate the source code **and** run automated code formatting:

```bash

> make generate

Reviewer: ```suggestion
$ make generate
```
